pkg/devfile/parser/types.go

```diff
@@ -134,7 +135,37 @@ func (d DevfileObj) OverrideProjects(overridePatch []common.DevfileProject) erro
 			}
 		}
 		if !found {
-			return fmt.Errorf("the command to override is not found in the parent")
+			return fmt.Errorf("the project to override is not found in the parent")
+		}
+	}
+	return nil
+}
+
+// OverrideStarterProjects overrides the starter projects of the parent devfile
+// overridePatch contains the patches to be applied to the parent's starter projects
+func (d DevfileObj) OverrideStarterProjects(overridePatch []common.DevfileStarterProject) error {
+	for _, patchProject := range overridePatch {
+		found := false
+		for _, originalProject := range d.Data.GetStarterProjects() {
+			if strings.ToLower(patchProject.Name) == originalProject.Name {
+				found = true
+				var updatedProject common.DevfileStarterProject
+
+				merged, err := handleMerge(originalProject, patchProject, common.DevfileStarterProject{})
+				if err != nil {
+					return err
+				}
+
+				err = json.Unmarshal(merged, &updatedProject)
+				if err != nil {
+					return err
```

Comment:
I think you should add a description to append to the error using `errors.Wrap(err, "unable to override starter project cannot unmarshal` for example. Since we use `json.Unmarshal` in other parts of the Devfile code when unmarshaling and it may be difficult to determine where the error is coming from.

Reply:
@cdrage json.Unmarshal specifies the field where the error is coming from 
https://play.golang.org/p/uQUjwN3H35G
```
json: cannot unmarshal string into Go struct field starterProject.ProjectNo of type int
```
I think the returned error gives sufficient information about the fields.
https://github.com/golang/go/blob/master/src/encoding/json/decode.go#L134

Reply:
Yes, but you should ideally wrap it as well since we use `json.Unmarshal` in other sections of our code. So having a clear indicator that it's coming from unmarshaling the starter project makes it easier to find where the error is coming from...

Reply:
Thanks, Done.